module/users/helper.js

```diff
@@ -0,0 +1,810 @@
+                    createdBy : userId,
+                    status : messageConstants.common.PUBLISHED
+                },[
+                    "entities",
+                    "solutionId",
+                    "programId",
+                    "entityId",
+                    "name",
+                    "description",
+                    "status",
+                    "observationId"
+                ]);
+
+                let observationIds = [];
+                let observationSolutions = [];
+                let observationEntities = [];
+                
+                if ( observationsData.length > 0 ) {
+                    
+                    observationsData.forEach(observation=>{
+                        observationIds.push(observation._id);
+                        observation["isObservation"] = true;
+                        programIds.push(observation.programId);
+                        observationSolutions.push(observation.solutionId);
+                        observationEntities = observationEntities.concat(observation.entities);
+                    })
+                }
+
+                let observationSubmissions = 
+                await observationSubmissionsHelper.observationSubmissionsDocument({
+                    observationId : { $in : observationIds },
+                    entityId : { $in : observationEntities }
+                },[
+                    "status",
+                    "submissionNumber",
+                    "entityId",
+                    "createdAt",
+                    "updatedAt",
+                    "observationInformation.name",
+                    "observationId", 
+                    "title",
+                    "completedDate",
+                    "ratingCompletedAt"   
+                ],{
+                    createdAt : -1
+                });
+
+                solutionIds = solutionIds.concat(observationSolutions);
+                entityIds = entityIds.concat(observationEntities);
+
+                if ( !programIds.length > 0 ) {
+                    throw {
+                        status : httpStatusCode.bad_request.status,
+                        message : messageConstants.apiResponses.PROGRAM_NOT_MAPPED_TO_USER
+                    }
+                }
+
+                if( !solutionIds.length > 0 ) {
+                    throw {
+                        status : httpStatusCode.bad_request.status,
+                        message : messageConstants.apiResponses.SOLUTION_NOT_MAPPED_TO_USER
+                    }
+                }
+                
+                if( !entityIds.length > 0 ) {
+                    throw {
+                        status : httpStatusCode.bad_request.status,
+                        message : messageConstants.apiResponses.ENTITIES_NOT_MAPPED_TO_USER
+                    }
+                }
+                
+                let programs = 
+                await programsHelper.programDocument(
+                    programIds,
```

Comment:
@aman-tunerlabs - Doesn't it need to be passed query with _id key ?

Reply:
@aks30 It was existing helper functionality which accept program ids only.

Reply:
OK